### ISSUE TITLE
I/O Development Task - Make it possible to dump ins/outs for a specific process

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#1303]
When debugging a process it is nice to be able to examine the set of fields going in and the set of fields coming out.  Currently in SCREAM the only way to do this is to run stand-alone, but when debugging full runs the best we can hope for is the set of initial conditions going in and the field values at the end of a single timestep.

This task would be to set up the infrastructure to write the full set of input to a process and the output from a process from within the context of a full simulation.  This could be a runtime flag added to the `input.yaml` file.

We may need to figure out how to five an atmosphere process the ability to write output directly.

**Comment by bartgol** [comment]
I think I want to prioritize this task. I am debugging the restart PR failure, which has only a handful of fields that diff. Being able to save them before/after each atm proc might allow to have a better idea of what's going on.

**Comment by AaronDonahue** [comment]
Yeah, this feature would really help with debugging right now.  Do you want to take on the development of the capability?

**Comment by bartgol** [comment]
Yeah, I already have a half idea in mind.
